config/plugins/sonic-hash.py

```diff
@@ -0,0 +1,273 @@
+"""
+This CLI plugin was auto-generated by using 'sonic-cli-gen' utility
+"""
+
+import click
+import utilities_common.cli as clicommon
+
+from sonic_py_common import logger
+from utilities_common.switch_hash import (
+    CFG_SWITCH_HASH,
+    STATE_SWITCH_CAPABILITY,
+    SW_CAP_HASH_FIELD_LIST_KEY,
+    SW_CAP_ECMP_HASH_KEY,
+    SW_CAP_LAG_HASH_KEY,
+    SW_HASH_KEY,
+    SW_CAP_KEY,
+    HASH_FIELD_LIST,
+    SYSLOG_IDENTIFIER,
+    get_param,
+    get_param_hint,
+    get_dupes,
+    to_str,
+)
+
+
+log = logger.Logger(SYSLOG_IDENTIFIER)
+log.set_min_log_priority_info()
+
+#
+# Hash validators -----------------------------------------------------------------------------------------------------
+#
+
+def hash_field_validator(ctx, param, value):
+    """
+    Check if hash field list argument is valid
+    Args:
+        ctx: click context
+        param: click parameter context
+        value: value of parameter
+    Returns:
+        str: validated parameter
+    """
+
+    for hash_field in value:
+        click.Choice(HASH_FIELD_LIST).convert(hash_field, param, ctx)
+
+    return list(value)
+
+
+def ecmp_hash_validator(ctx, db, ecmp_hash):
+    """
+    Check if ECMP hash argument is valid
+
+    Args:
+        ctx: click context
+        db: State DB connector object
+        ecmp_hash: ECMP hash field list
+    """
+
+    dup_list = get_dupes(ecmp_hash)
+    if dup_list:
+        raise click.UsageError("Invalid value for {}: {} has a duplicate hash field(s) {}".format(
+            get_param_hint(ctx, "ecmp_hash"), to_str(ecmp_hash), to_str(dup_list)), ctx
+        )
+
+    entry = db.get_all(db.STATE_DB, "{}|{}".format(STATE_SWITCH_CAPABILITY, SW_CAP_KEY))
```

Comment:
can we enhance it for multi-asic ?

Reply:
@abdosi multi ASIC is out of scope according to HLD